notebooks/experiment_storage_penetration.jl

```diff
@@ -0,0 +1,329 @@
+	Random.seed!(2)
+	renew_profiles = rand(0:n_renew, n)
+	
+	
+	get_renew = (t, renew_profile) -> 
+		renew_profile == 0 ? 0.0 : renew_data[t, renew_profile]
+	
+	d_dyn = [
+		d_peak .* (demand_data[t, demand_profiles] - renewable_pentration * get_renew.(t, renew_profiles))
+		for t in 1:T
+	]
+end;
+
+# ╔═╡ bfa4a8f9-cfcd-4e22-b2dc-751226f3a73c
+begin
+	Random.seed!(4)
+	plot(size=(600, 200))
+	plot!(hcat(d_dyn...)'[:, rand(1:n, 10)], lw=2, c=repeat(renew_profiles, outer=T)')
+end
+
+# ╔═╡ a8ccbc8e-24e6-4214-a179-4edf3cf26dad
+md"""
+### Carbon emissions data
+"""
+
+# ╔═╡ 897cc3a0-66cb-417d-81d5-20036b32dd99
+net.gmax
+
+# ╔═╡ e8ee5cbb-4afc-4737-b006-90071f6138cd
+begin
+	Random.seed!(3)
+	emission_rates = 1000 .+ 1000*rand(length(net.gmax))
+end;
+
+# ╔═╡ 30ec492c-8d21-43f6-bb09-32810494f21e
+md"""
+## How does storage penetration affect MEFs?
+"""
+
+# ╔═╡ 64b06f05-b1ab-44c4-80e2-faeeb1466421
+c_rate = 0.25
+
+# ╔═╡ 98a0d7c5-b1a8-4ebe-bb73-7ca88b475592
+# TODO Change calculation for penetration
+storage_penetrations = [0.0, 0.5, 1.0]
+
+# ╔═╡ 8fc06205-0227-4b46-a2e9-72bdf9d57926
+mef_times = 2:23
+
+# ╔═╡ 008ed573-67ec-4908-a51a-c5d2a01e5b0e
+refresh = false
+
+# ╔═╡ 3706fd91-dd00-48bc-920e-b6bbc413352c
+
+
+# ╔═╡ 52c6bcd9-1f6c-40a3-89c0-5c95b4d58421
+
+
+# ╔═╡ 6010defa-fc4c-4cd9-8450-3921d778c735
+
+
+# ╔═╡ 418861e0-a35e-47db-9f00-a6a7fcf733fe
+md"""
+## Appendix
+"""
+
+# ╔═╡ 0d07df73-89e1-4dbf-8f8c-82c202ad84c7
+md"""
+### Solve static problem
+
+Solve the static optimal power flow problem and display the LMPs.
+"""
+
+# ╔═╡ 43ab37f7-bf1f-44fb-8858-e3bf7d4e8880
+begin
+	for t in 1:T
+		opf_t = PowerManagementProblem(net, d_dyn[t])
+		solve!(opf_t, OPT, verbose=true)
+		@assert opf_t.problem.status == Convex.MOI.OPTIMAL
+	end
+		
+	opf = PowerManagementProblem(net, d_dyn[20])
+	solve!(opf, OPT, verbose=true)
+	
+	opf.problem.status
+end
+
+# ╔═╡ e1ef0db3-3130-45b0-9f07-c5776d72c31a
+begin
+	λ = get_lmps(opf)
+	bar(λ, size=(600, 200), ylim=(100, Inf), title="locational marginal prices")
+end
+
+# ╔═╡ 08cce787-8118-4792-829a-153d2b637a78
+bar(abs.(evaluate(opf.p)) ./ (net.pmax), size=(600, 100), title="line flows")
+
+# ╔═╡ 7e1bec31-9cdf-466c-83b3-dc792fd5cc53
+md"""
+### Utilities
+"""
+
+# ╔═╡ a3105ef4-67e7-41ad-bd2e-5d458c853d80
+function make_dynamic(net, T, P, C)
+	fqs = [net.fq for t in 1:T]
+	fls = [net.fl for t in 1:T]
+	pmaxs = [net.pmax for t in 1:T]
+	gmaxs = [net.gmax for t in 1:T]
+	return DynamicPowerNetwork(fqs, fls, pmaxs, gmaxs, net.A, net.B, P, C)
+end
+
+# ╔═╡ 6f08828b-4c4b-4f50-bd40-35805a37aae0
+begin
+	# Recompute results
+	if refresh || !isfile("../results/storage.jld")
+		options = (
+			c_rate=c_rate,
+			storage_penetrations=storage_penetrations,
+			mef_times=mef_times,
+			emission_rates=emission_rates,
+			d_dyn=d_dyn,
+		)
+		
+		results = zeros(n, length(mef_times), length(storage_penetrations))
+		for (ind_s, s_rel) in enumerate(storage_penetrations)
+
+			# Construct dynamic network
+			C = d_peak * s_rel
```

Comment:
The definition of storage penetration is therefore a scaling wrt the peak demand? That makes sense i believe. 

In this case, there is no reason to be up-limited by `1`, correct?